### DISCUSSION TITLE
一些重要的小建议

### DISCUSSION THREAD

**Discussion by cosmicethics** [nshen/learn-neovim-lua#49] [General]
# 小建议

前不久刚用的nvim，从作者的配置一步一步修改过来，在此非常感谢作者（我没买小册子，白嫖配置😂），让我不至于一头雾水扎进茫茫多的配置中。自己刚开始使用时，觉得很多插件用不着就删除了，后来又陆陆续续下载回来了，看来作者的配置真的足够合理，在这途中也发现了一些可以优化的地方：

1. lsp-installer和treesitter install都可以设置镜像(fastgit)

     - plugin-config/nvim-treesitter.lua 开头部分添加:

        ```lua
        -- 设置镜像
        for _, config in pairs(require("nvim-treesitter.parsers").get_parser_configs()) do
            config.install_info.url = config.install_info.url:gsub("https://github.com/", "https://hub.fastgit.xyz/")
        end

        ```

     - lsp/setup.lua setup()里面添加：

       ```lua
        github = { 
            -- The template URL to use when downloading assets from GitHub.
            -- The placeholders are the following (in order):
            -- 1. The repository (e.g. "rust-lang/rust-analyzer")
            -- 2. The release version (e.g. "v0.3.0")
            -- 3. The asset name (e.g. "rust-analyzer-v0.3.0-x86_64-unknown-linux-gnu.tar.gz")
            download_url_template = "https://download.fastgit.org/%s/releases/download/%s/%s",
        },
       ```

2. bufferline ui优化：最左边的buffer分隔符会卡到到nvim-tree里面，只需添加padding

    ```diff
        offsets = {
            {
                filetype = "NvimTree",
                text = "File Explorer",
                highlight = "Directory",
                text_align = "left",
    +           padding = 1,
            },
        }
    ```

3. lualine-lsp-progress和fidget.nvim功能重复了，建议只留下fidget

4. 不常用的功能，不要设置按键映射，使用:command就好了。

5. 边用边优化中

**Comment by nshen** [comment]
感谢建议，设置镜像好像很实用。

试了一下padding，我觉得 0 也挺好的

- bufferline padding = 1
<img width="510" alt="image" src="https://user-images.githubusercontent.com/181506/176986963-996178fe-428e-4bf0-90a7-96ebbab057cd.png">

- bufferline padding = 0
<img width="537" alt="image" src="https://user-images.githubusercontent.com/181506/176986979-26755670-7830-4045-941f-dbf45dad893a.png">
